maint/qmpi_binding.py

```diff
@@ -265,38 +266,84 @@ def generate_code(parseset: Mapping[str, Any], postfix: str, kind_map: Mapping[s
 enum_list_string = (',\n    '.join(enum_list));
 qmpi_h_file.write(f"enum QMPI_Functions_enum {{" "\n    "
                   f"{enum_list_string}," "\n"
+                   "MPI_FILE_C2F_T,\n"
+                   "MPI_FILE_F2C_T,\n"
                    "MPI_LAST_FUNC_T\n"
                   f"}};" "\n\n");
 
 qmpi_h_file.write(""
         "typedef struct {\n"
-        "    void *storage_stack;\n"
+        "    void **storage_stack;\n"
+        "    void *calling_address;\n"
         "} QMPI_Context;\n"
         "\n"
         );
 
 qmpi_h_file.write(''.join(qmpi_h_file_text));
 
+qmpi_h_file.write(
+    "MPI_Fint QMPI_File_c2f(QMPI_Context context, int tool_id, MPI_File fh) MPICH_API_PUBLIC;\n"
+    "typedef MPI_Fint (QMPI_File_c2f_t) (QMPI_Context context, int tool_id, MPI_File fh);\n"
+    "\n"
+    "MPI_File QMPI_File_f2c(QMPI_Context context, int tool_id, MPI_Fint fh) MPICH_API_PUBLIC;\n"
+    "typedef MPI_File (QMPI_File_f2c_t) (QMPI_Context context, int tool_id, MPI_Fint fh);\n"
+    "\n"
+);
+
```

Comment:
What was going on with these functions that they couldn't be written automatically?

Reply:
I think, as discussed before they are not part of the mpi-standard. They are part of the romio src tree. I am not able to find any PR for it.

Reply:
They're definitely in the MPI Standard, but they might have been missed in the Pythonization code. I'll have to go take a look.

Reply:
It looks like this PR has still not been merged and is necessary for this to show up: https://github.com/mpi-forum/mpi-standard/pull/276